lightning/src/ln/channel.rs

```diff
@@ -152,6 +152,50 @@ enum InboundHTLCState {
 	LocalRemoved(InboundHTLCRemovalReason),
 }
 
+#[derive(Clone, Debug, PartialEq)]
+pub enum InboundHTLCStateDetails {
+	RemoteAnnouncedForward,
+	RemoteAnnouncedFail,
```

Comment:
I kept this state around as otherwise the pending HTLC would be dropped.

Reply:
I think I mainly wanted the state details to be a `.map()` of internal state instead of a `.filter_map()`. Added documentation around `RemoteAnnounced` states.

Reply:
One of the reasons for adding this PR was also observability, so I was leaning towards exposing information we have, especially since the states are caused by and are dependent on external messages. It might for example be possible to get stuck at the `RemoteAnnounced` state if the counterparty does not send `commitment_signed` due to an implementation bug.

I agree that information on announcements adds less value than the rest though and I probably should be consistent with https://github.com/lightningdevkit/rust-lightning/pull/2442#discussion_r1287734864 as well, added a commit to not expose this.